pkg/sstable/sstable_naming.go

```diff
@@ -0,0 +1,149 @@
+	mulLow  = 4865540595714422341
+	mulHigh = 2549297995355413924
+)
+
+// ExtractID returns ID from SSTable name in a form string integer or UUID:
+// me-3g7k_098r_4wtqo2asamoc1i8h9n-big-CRC.db -> 3g7k_098r_4wtqo2asamoc1i8h9n
+// me-7-big-TOC.txt -> 7
+// Supported SSTable format versions are: "mc", "md", "me", "la", "ka".
+// Scylla code validating SSTable format can be found here:
+// https://github.com/scylladb/scylladb/blob/decbc841b749d8dd3e2ddd4be4817c57d905eff2/sstables/sstables.cc#L2115-L2117
+func ExtractID(sstable string) string {
+	parts := strings.Split(sstable, "-")
+
+	if regexLaMx.MatchString(sstable) || regexNewLaMx.MatchString(sstable) {
+		return parts[1]
+	}
+	if regexKa.MatchString(sstable) {
+		return parts[3]
+	}
+
+	panic(unknownSSTableError(sstable))
+}
+
+func replaceID(sstable, newID string) string {
+	parts := strings.Split(sstable, "-")
+
+	switch {
+	case regexLaMx.MatchString(sstable):
+		parts[1] = newID
+	case regexKa.MatchString(sstable):
+		parts[3] = newID
+	default:
+		panic(unknownSSTableError(sstable))
+	}
+
+	return strings.Join(parts, "-")
+}
+
+func unknownSSTableError(sstable string) error {
+	return errors.Errorf("unknown SSTable format version: %s. Supported versions are: 'mc', 'md', 'me', 'la', 'ka'", sstable)
+}
+
+// RenameToIDs reformat sstables ids to ID format keeping id consistency and uniqueness.
+func RenameToIDs(sstables []string, globalCounter *atomic.Int64) map[string]string {
+	return RenameSStables(sstables, func(_ string) string {
+		return strconv.Itoa(int(globalCounter.Inc()))
+	})
+}
+
+// RenameToUUIDs reformat sstables ids to UUID format keeping id consistency.
+func RenameToUUIDs(sstables []string) map[string]string {
+	return RenameSStables(sstables, func(id string) string {
+		if !regexLaMx.MatchString(id) && regexNewLaMx.MatchString(id) {
+			return id
+		}
+		return RandomSSTableUUID()
+	})
+}
+
+// RenameSStables resolves sstable file name conflicts replacing id in the names with unique id value.
+// SSTables originating from different nodes could have identical names, leading to the conflict in which one overwrites another during download.
+// To avoid that we need to remap them to unique names.
+func RenameSStables(sstables []string, nameGen func(id string) string) map[string]string {
+	idMapping := make(map[string]string)
+	out := make(map[string]string)
+
+	for _, sst := range sstables {
+		id := ExtractID(sst)
+		newID, ok := idMapping[id]
+		if !ok {
+			newID = nameGen(id)
+			idMapping[id] = newID
+		}
+
+		out[sst] = replaceID(sst, newID)
+	}
+
+	return out
+}
+
+// RandomSSTableUUID generates random sstable uuid in a form of `497z_213k_3zpaqrwk2na921344z`.
+func RandomSSTableUUID() string {
+	var buff [26]byte
+	_, err := io.ReadFull(rand.Reader, buff[:])
+	if err != nil {
+		pseudoRandom.Read(buff[:])
+	}
+	var out [26]byte
+	for id, val := range buff {
+		out[id] = letters[val%36]
+	}
+	return string(out[0:4]) + "_" + string(out[4:8]) + "_" + string(out[8:])
+}
```

Comment:
Maybe it would be good to include a link to an actual place in Scylla where those IDs are generated? This line looks suspicious without any context:
```go
return string(out[0:4]) + "_" + string(out[4:8]) + "_" + string(out[8:])
```
https://github.com/scylladb/scylladb/blob/decbc841b749d8dd3e2ddd4be4817c57d905eff2/sstables/generation_type.hh#L68

Reply:
Added